BEST_PRACTICES.md

```diff
@@ -226,88 +218,74 @@ In this way we are sure only a single instance of the client is created and reus
-const authMiddlewareOptions = {}
-const httpMiddlewareOptions = {}
-
-const client = new ClientBuilder()
-  .withAuthMiddlewareOption(authMiddlewareOptions)
-  .withHttpMiddlewareOption(httpMiddlewareOptions)
-  ...
-  .build()
-```
+The Commercetools JS/TS client can be customized in a variety of ways, this can be achieved by passing in middleware options that can take different parameter values.
+Refer to the [Client Initialization](#client-initialization) section for details on how to create the client.
+To see how to configure the available middleware options and their configurations, see [this documentation](https://commercetools.github.io/nodejs/sdk/api/#middlewares).
 
 ## Error Handling
 
-Internally the JS/TS SDK implements some helper functions that are connected with error handling, given an error code and message, an error object can be constructed.
+The Commercetools JS/TS SDK provides helper functions for handling API errors.
+`getErrorByCode` retrieves a constructor function (a class) for a specific HTTP error type (for example 400 Bad Request, 404 Not Found). You then use this constructor with `new` to create an instance of the error.
 
 ```ts
-import { getErrorByCode } from '@commercetools/sdk-client-v2'
-
-const ErrorType = getErrorByCode(400)
-const error = new ErrorType('Oops this is an error')
-
-// error is of type
-type HttpErrorType = {
-  name: string
-  message: string
-  code: number
-  status: number
-  statusCode: number
-  body: Object
-  originalRequest: ClientRequest
-  headers?: {
-    [key: string]: string
+import { getErrorByCode, HttpErrorType } from '@commercetools/ts-client'
+
+try {
+  const BadRequestError = getErrorByCode(400)
+
+  if (BadRequestError) {
+    throw new BadRequestError('Invalid request data')
+  }
+} catch (error) {
+  if (error instanceof BadRequestError) {
+    console.error('Bad Request:', error.message, error.statusCode)
+  } else if (error instanceof Error) {
+    console.error('Generic Error:', error.message)
+  } else if (error as HttpErrorType) {
+    console.error('HTTP Error:', error.message, error.statusCode, error.body)
+  } else {
+    console.error('Unknown error:', error)
   }
 }
 ```
 
 ## Configuring proxies
 
-Proxies can be configured at the http client level in the JS/TS SDK, here we pass the proxy url/ip address
+Proxies can be configured at the HTTP client level in the Commercetools JS/TS SDK, here we pass the proxy url/ip address
 
 ```ts
 import HttpsProxyAgent from 'https-proxy-agent'
 
 const fetcherProxy = (url, fetchOptions = {}) => {
-   fetchOptions.agent = new HttpsProxyAgent('proxy-url/ip-address') // http://76.253.101.51:8080
-   return fetch(url, fetchOptions)
+  fetchOptions.agent = new HttpsProxyAgent('proxy-url/ip-address') // http://76.253.101.51:8080
+  return fetch(url, fetchOptions)
 }
 
 const httpMiddlewareOptions = {
-  ...
-  fetch: fetcherProxy
-  ...
+  //...
```

Comment:
```suggestion
  ...
```

Reply:
I believe that just this it's better to be commented otherwise can be confused with Spread operator or Rest parameters.